receiver/mongodbatlasreceiver/events.go

```diff
@@ -228,6 +228,11 @@ func (er *eventsReceiver) transformEvents(now pcommon.Timestamp, events []*mongo
 }
 
 func (er *eventsReceiver) checkpoint(ctx context.Context) error {
+	// if storageClient was never set, we shouldn't try to checkpoint
+	if er.storageClient == nil {
```

Comment:
If `Start` completes successfully, this is never nil, so we only need to check this during shutdown.

Reply:
Alternately, we can initialize the client to `storage.NewNowClient` and not have to worry about the check at all.